internal/store/http-fetch.go

```diff
@@ -0,0 +1,127 @@
+package store
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type RemoteImageList struct {
+	BaseURL string
+}
+
+type TagListResponse struct {
+	Name string   `json:"name"`
+	Tags []string `json:"tags"`
+}
+
+func RemoteImageListFetcher(url string) *RemoteImageList {
+	return &RemoteImageList{
+		BaseURL: url,
+	}
+}
+
+func (r *RemoteImageList) Type() string {
+	return "Remote"
+}
+
+func (client *RemoteImageList) List(ctx context.Context) ([]Image, error) {
+	// Construct the URL for fetching tags
+	url := client.BaseURL + "/tags/list"
+
+	// Encode credentials for Basic Authentication
+	username := os.Getenv("HARBOR_USERNAME")
+	password := os.Getenv("HARBOR_PASSWORD")
+	auth := base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+
+	// Create a new HTTP request
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Set the Authorization header
+	req.Header.Set("Authorization", "Basic "+auth)
+
+	// Send the request
+	httpClient := &http.Client{}
```

Comment:
Create an http client for the struct rather than creating one for every function call.